sycl/doc/extensions/ExplicitSIMD/dpcpp-explicit-simd.md

```diff
@@ -0,0 +1,776 @@
+the selected sub-vector starting from the i-th element. The number of selected
+elements is specified by the template parameter **Size**, while the distance
+between two adjacent elements is specified by the template parameter **Stride**.
+
+```cpp
+  simd<int, 8> a;
+  simd<int, 4> b;
+  // ...
+  b = a.select<4, 2>(1);  // size=4, stride=2, offset=1 (elements a(1),
+                          // a(3), a(5) and a(7) are copied to b)
+```
+
+<p align="center">
+<img src="images/VectorOdd.svg" title="1D select example" width="600" height="60"/>
+</p>
+
+```cpp
+a.select<4, 2>(0) = b;  // selected elements of a are replaced
+                        // with elements of b (all elements of b are
+                        // copied to elements a(0), a(2), a(4), a(6))
+```
+
+<p align="center">
+<img src="images/VectorEven.svg" title="1D select example" width="600" height="60"/>
+</p>
+
+
+Gen ISA provides powerful register region addressing modes to facilitate cross-lane
+SIMD vector operation. To exploit this feature Explicit SIMD provides ```replicate``` function
+to allow programmer to implement any native Gen ISA region in the following forms:
+- ```replicate<REP>()```: replicate a simd vector object **REP** times and return a new simd
+vector of **REP** * Width, where Width specifies the original vector size.
+- ```replicate<REP, W>(uint16_t i)```: replicate **W** consecutive elements starting at the
+i-th element from the simd vector object **REP** times, and return a new simd vector of **REP** * **W** length.
+- ```replicate<REP, VS, W>(uint16_t i)```: replicate **REP** blocks of **W** consecutive
+elements starting at the i-th from the simd vector object with each block strided by **VS**
+elements, and return a new vector of **REP** * **W** length. Selected blocks of **W**
+elements will overlap if **VS** < **W**.
+- ```replicate<REP, VS, W, HS>(uint16_t i=0 )```: replicate **REP** blocks of **W** sequential
+elements with a stride of **HS** starting at the i-th element from the simd vector object with
+each block strided by **VS** elements, and return a new vector of **REP** * **W** length.
+Selected blocks of **W** elements will overlap if **VS** < **W**.
+
+To avoid explicit type cast and the resulting move instructions for large vectors, Explicit SIMD allows
+programmer to reinterpret the fundamental data element type of a simd vector object and change
+its shape to 1D or 2D object through the ```format``` function:
+- ```format<EltTy>( )```: returns a reference to the calling simd object interpreted as a new
+simd vector with the size determined by the template **EltTy** parameter.
+- ```format<EltTy, Height, Width>( )```: returns a reference to the calling simd object interpreted
+as a new 2D simd_view object with the shape determined by the template parameters **Height** and**Width**. The size of the new 2D block must not exceed the size of the original object.
+
+```cpp
+  simd<int, 16> v1;
+   // ...
+  auto v2 = v1.format<short>;
```

Comment:
It sounds like a specialization of the `select` above.
And one can ask how it relates to `std::format`...

Reply:
It is what C++20 calls `std::bit_cast`, not the select. Select takes subset of values, while format - bit_casts them (into potentially differently sized set) and reshapes them (but in overloads with more template arguments).

By the way, format is a function - where are parentheses?

Reply:
Why not call it bit_cast then?

Reply:
added TODO to the headers